src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Data.cs

```diff
@@ -17,11 +17,98 @@ namespace Microsoft.NetCore.Analyzers.InteropServices
 {
     using ValueContentAnalysisResult = DataFlowAnalysisResult<ValueContentBlockAnalysisResult, ValueContentAbstractValue>;
 
-    public sealed partial class RuntimePlatformCheckAnalyzer
+    public sealed partial class PlatformCompatabilityAnalyzer
     {
-        private struct RuntimeOSPlatformInfo : IAbstractAnalysisValue, IEquatable<RuntimeOSPlatformInfo>
+        private enum PlatformAttributeType
         {
-            private RuntimeOSPlatformInfo(string invokedPlatformCheckMethodName, string platformPropertyName, Version version, bool negated)
+            None, MinimumOSPlatformAttribute, ObsoletedInOSPlatformAttribute, RemovedInOSPlatformAttribute, TargetPlatformAttribute
+        }
+
+        private struct PlatformAttributeInfo : IEquatable<PlatformAttributeInfo>
+        {
+            public PlatformAttributeType AttributeType { get; set; }
+            public string PlatformName { get; set; }
+            public Version Version { get; set; }
+
+            public static bool TryParsePlatformAttributeInfo(AttributeData osAttribute, out PlatformAttributeInfo parsedAttribute)
+            {
+                parsedAttribute = new PlatformAttributeInfo();
+                switch (osAttribute.AttributeClass.Name)
+                {
+                    case MinimumOSPlatformAttribute:
+                        parsedAttribute.AttributeType = PlatformAttributeType.MinimumOSPlatformAttribute;
+                        break;
+                    case ObsoletedInOSPlatformAttribute:
+                        parsedAttribute.AttributeType = PlatformAttributeType.ObsoletedInOSPlatformAttribute;
+                        break;
+                    case RemovedInOSPlatformAttribute:
+                        parsedAttribute.AttributeType = PlatformAttributeType.RemovedInOSPlatformAttribute;
+                        break;
+                    case TargetPlatformAttribute:
+                        parsedAttribute.AttributeType = PlatformAttributeType.TargetPlatformAttribute;
+                        break;
+                    default:
+                        parsedAttribute.AttributeType = PlatformAttributeType.None;
+                        break;
+                }
+
+                if (osAttribute.ConstructorArguments.Length == 1 && osAttribute.ConstructorArguments[0].Type.SpecialType == SpecialType.System_String &&
+                    !osAttribute.ConstructorArguments[0].IsNull && TryParsePlatformNameAndVersion(osAttribute.ConstructorArguments[0].Value.ToString(), out string platformName, out Version? version))
```

Comment:
Do you need to check `Kind` for the argument is not `Array` before accessing the `Value` to avoid the exception here: http://sourceroslyn.io/#Microsoft.CodeAnalysis/Symbols/TypedConstant.cs,97

Reply:
No, the attribute doesn't have overload accepting an array

Reply:
The check would be guarding against invalid user input.

Reply:
Isn't compiler would warn that attribute have no overload with array?

Reply:
Can we add a test to confirm where user provides an array input and there is a compiler semantic error?

Reply:
I can add a test, but don't think such test related to the analyzer scope

Reply:
Yeah, just a alternate way to mimic different reference assemblies with identical internal attribute definitions.